### PR TITLE
"Element language" rules: handle whitespace

### LINKED ISSUE #1693: "Element with lang attribute has valid language tag" and "HTML element language subtag matches language" [de46e4, off6ek] should ignore whitespace only text
```
<body><div lang="invalid"> <span lang="en">Hello</span> </div></body>
```
[Element with lang attribute has valid language tag](https://act-rules.github.io/rules/de46e4) fails this because there are two non-empty text nodes within the `div`, but these are whitespace and don't create problem.

[HTML element language subtag matches language](https://act-rules.github.io/rules/off6ek) is a bit mixed up there, these whitespace text nodes do not have any [most common language](https://act-rules.github.io/rules/off6ek#most-common-element-language) due to having no words, which make the expectation fails (`invalid` is not a most common language).

Suggestions (choose one):
1. In both rules, replace the last Applicability bullet from "non-empty text" to "non-whitespace text"
    **Problem**: we're still running into the same issue with corner cases of non-text "text" (e.g. smiley, punctuation, …); OTOH I'm not sure how this is announced… But at least `<span lang="en">😀</span>` doesn't have a most common language and therefore fails [HTML element language subtag matches language](https://act-rules.github.io/rules/off6ek). (yes, it's an Unicode smiley inside that text node, also works with an ASCII art one like `:-)`)
2. In both rules, update Expectation to say something like "either the element [on which the attribute is declared] has no most common language, or [current check]"
    **Problem**: this makes [Element with lang attribute has valid language tag](https://act-rules.github.io/rules/de46e4) much harder to fully automate since it needs to know the most common language. I sorta assume that tools will default to heuristic like "non-whitespace text". Depending on how strict we build our tests cases, this may result in having less implementations for the rule 😢 
3. Update Applicability of [Element with lang attribute has valid language tag](https://act-rules.github.io/rules/de46e4) to reject whitespace text nodes; and Expectation of [HTML element language subtag matches language](https://act-rules.github.io/rules/off6ek) to explicitly pass if there are no most common language.
    **Problem?**: [Element with lang attribute has valid language tag](https://act-rules.github.io/rules/de46e4) would still fail `<span lang="invalid">😀</span>` (or `<span lang="invalid">:-)</span>`). But maybe that is OK because the emojis (or symbols) are actually announced and therefore screen readers need a language to pick (i.e. "smiling face" is not the same as "visage souriant", so having an invalid `lang` attribute on such a `span` is indeed a real problem) 🤔

### PR DESCRIPTION
**Pull request by Jym77**
On both rules about elements' `lang`:
* make whitespace only text inapplicable
* add an assumption that the text contain human language (clears up corner cases of text which would not have a "most common language" and similar stuff).

Closes issue(s):

- closes #1693 

Need for Call for Review:
This will require a 1 week Call for Review (updating Applicability)

---

### **When creating PR:**

- [X] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).
- [X] Make sure you do not remove the "How to Review and Approve" section in your pull request description

### **After creating PR:**

- [X] Add yourself (and co-authors) as "Assignees" for PR.
- [X] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [X] [Link the PR to any issue it solves](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). This will be done automatically by [referencing the issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) at the top of this comment in the indicated place.
- [X] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

### **When merging a PR:**

- [ ] Close any issue that the PR resolves. This will happen automatically upon merging if the PR was correctly linked to the issue, e.g. by referencing the issue at the top of this comment.

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Call for Review period. In case of disagreement, the longer period wins.

### CODE REVIEW

File: _rules/element-lang-matches-default-language-off6ek.md [hunk 1]
```diff
@@ -325,11 +327,25 @@ The `lang` [attribute value][] of this `p` element is a [grandfathered tag][gran
 </html>
 ```
 
+#### Inapplicable Example 8
+
+The [text inheriting its programmatic language][] from this `div` element it only [whitespace][].
+
+```html
+<html>
```
Comment (WilcoFiers) [inline _rules/element-lang-matches-default-language-off6ek.md:335]: I don't think SC 3.1.1 allows this TBH. There is no "except if the page is empty" or "except if the page has no default language" exception in that SC. The other rule I agree with, but I don't think we should do the same to this rule.

**Review by WilcoFiers — CHANGES_REQUESTED** [review]

File: _rules/element-lang-matches-default-language-off6ek.md (hunk 1 above)
Comment (WilcoFiers) [inline _rules/element-lang-matches-default-language-off6ek.md:335]: Needs a lang attribute :)

File: _rules/element-lang-matches-default-language-off6ek.md [hunk 2]
```diff
@@ -325,11 +327,25 @@ The `lang` [attribute value][] of this `p` element is a [grandfathered tag][gran
 </html>
 ```
 
+#### Inapplicable Example 8
+
+The [text inheriting its programmatic language][] from this `div` element it only [whitespace][].
```
Comment (carlosapaduarte) [inline _rules/element-lang-matches-default-language-off6ek.md:332]:
```suggestion
The [text inheriting its programmatic language][] from this `div` element is only [whitespace][].
```

File: _rules/element-lang-valid-de46e4.md [hunk 3]
```diff
@@ -322,15 +324,30 @@ There is no [text inheriting its programmatic language][] from this `div` elemen
 </html>
 ```
 
+#### Inapplicable Example 5
+
+The [text inheriting its programmatic language][] from this `div` element it only [whitespace][].
```
Comment (carlosapaduarte) [inline _rules/element-lang-valid-de46e4.md:329]:
```suggestion
The [text inheriting its programmatic language][] from this `div` element is only [whitespace][].
```

**Review by carlosapaduarte — CHANGES_REQUESTED** [review]
Found a typo :-)

**Review by carlosapaduarte — APPROVED** [review]

File: _rules/element-lang-matches-default-language-off6ek.md [hunk 4]
```diff
@@ -325,11 +327,25 @@ The `lang` [attribute value][] of this `p` element is a [grandfathered tag][gran
 </html>
 ```
 
+#### Inapplicable Example 8
+
+The [text inheriting its programmatic language][] from this `div` element is only [whitespace][].
```
Comment (WilcoFiers) [inline _rules/element-lang-matches-default-language-off6ek.md:332]: I'm not following... the div is empty. It has no whitespace. Did something go wrong with Prettier or something?

**Review by WilcoFiers — CHANGES_REQUESTED** [review]

File: _rules/element-lang-matches-default-language-off6ek.md (hunk 4 above)
Comment (Jym77) [reply _rules/element-lang-matches-default-language-off6ek.md:332]: Yes, Prettier seems to remove them 😢 
Do you have an idea how to circumvent that?

(my current best would be to add some JS to do `foo.innerText = "    "` but that seems a bit too crazy…)

**Review by WilcoFiers — APPROVED** [review]

**Review by kherr-a11y — APPROVED** [review]

**Comment by Jym77** [comment]
Call for Review ends on February 10.

**Review by colabottles — APPROVED** [review]

**Comment by Jym77** [comment]
Call for Review has ended, merging.

**Pull request merged**